Meadowlark-js/packages/meadowlark-core/src/metaed/MetaEdValidation.ts

```diff
@@ -77,16 +77,57 @@ export function validateQueryParametersAgainstSchema(
 ): string[] {
   // TODO: RND-67 removed the Joi-specific code to report invalid query parameters
   // RND-307 will restore as a part of understanding Ajv error metadata
+
+  let errors: string[] = [];
+
   const schema = {
     ...metaEdModel.data.meadowlark.jsonSchema,
     // Need to relax the validation such that no fields are "required"
     required: [],
   };
 
+  const parameterKeys = Object.keys(queryParameters);
+
+  /**
+   * Number and boolean are come through as strings, which causes ajv to fail validation of parameters
+   * This loops through all query parameters and checks the type against the schema, if the type is numeric or boolean
+   * it will attempt convert the value. If the conversion is good, we set the value back and ajv will validate the schema
+   * If the value can't be converted (i.e. a word is provided for a numeric value, we leave it and let ajv invalidate
+   * the schema).
+   */
+  // eslint-disable-next-line no-restricted-syntax
+  for (const keyValue of parameterKeys) {
+    const property = schema.properties[keyValue];
+
+    if (property != null && property.type != null) {
+      if ((property && property?.type === 'integer') || property.type === 'number') {
+        const value = Number(queryParameters[keyValue]);
+        if (!Number.isNaN(value)) {
+          queryParameters[keyValue] = value;
+        }
+      }
+      if (property.type === 'boolean' && (queryParameters[keyValue] === 'true' || queryParameters[keyValue] === 'false')) {
+        queryParameters[keyValue] = Boolean(queryParameters[keyValue]);
+      }
+    }
+  }
+
   const valid: ValidateFunction = ajv.compile(schema);
 
-  const isValid: boolean = valid(queryParameters);
+  valid(queryParameters);
```

Comment:
This is a strange object: `valid` is both a function call and it has properties (i.e. `errors`). Not much we can do about that.

I liked the older pattern of having `isValid` and `if (isValid) return [];` because it provides a "fast" response to the positive situation, and then the negative situation is at the root of the function rather than being indented. I just find it a little easier to read when the "quick path" returns first before the code path that needs to do more work to create the return value.

Reply:
This was a remnant from when I thought I was going to handle the errors from the validation code above myself, didn't need to and didn't revert back to this style, updated.